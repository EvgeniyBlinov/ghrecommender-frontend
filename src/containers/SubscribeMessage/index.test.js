import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { PromiseState } from 'react-refetch'

import { SubscribeMessageContainer } from './index'

const props = {
  postSubscribe: jest.fn(),
  subscribed: false,
}

describe('<SubscribeMessageContainer />', () => {
  test('loading', () => {
    const postSubscribeResponse = PromiseState.create()
    const component = renderer.create(
      <SubscribeMessageContainer {...props} postSubscribeResponse={postSubscribeResponse} />,
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('subscribed: false', () => {
    const component = renderer.create(<SubscribeMessageContainer subscribed={false} />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('subscribed: true', () => {
    const component = renderer.create(<SubscribeMessageContainer subscribed />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('error', () => {
    const postSubscribeResponse = PromiseState.reject()
    const component = renderer.create(
      <SubscribeMessageContainer {...props} postSubscribeResponse={postSubscribeResponse} />,
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('success', () => {
    const postSubscribeResponse = PromiseState.resolve({ status: true })
    const component = renderer.create(
      <SubscribeMessageContainer {...props} postSubscribeResponse={postSubscribeResponse} />,
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('Click', () => {
    const component = mount(<SubscribeMessageContainer {...props} />)
    component.find('Button').simulate('click')
    expect(props.postSubscribe.mock.calls.length).toBe(1)
  })
})
