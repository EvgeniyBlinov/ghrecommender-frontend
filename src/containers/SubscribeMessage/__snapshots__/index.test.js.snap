// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscribeMessageContainer /> error 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="mail outline icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Subscribe to get new recommendations as soon as they arrive.
    </div>
    <br />
    <button
      className="ui small positive button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Subscribe
    </button>
  </div>
</div>
`;

exports[`<SubscribeMessageContainer /> loading 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="circle notched loading icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Subscribe to get new recommendations as soon as they arrive.
    </div>
    <br />
    <button
      className="ui small loading positive button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Subscribe
    </button>
  </div>
</div>
`;

exports[`<SubscribeMessageContainer /> subscribed: false 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="mail outline icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Subscribe to get new recommendations as soon as they arrive.
    </div>
    <br />
    <button
      className="ui small positive button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Subscribe
    </button>
  </div>
</div>
`;

exports[`<SubscribeMessageContainer /> subscribed: true 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="mail outline icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Subscribe to get new recommendations as soon as they arrive.
    </div>
    <br />
    <button
      className="ui small negative button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Unsubscribe
    </button>
  </div>
</div>
`;

exports[`<SubscribeMessageContainer /> success 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="mail outline icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Subscribe to get new recommendations as soon as they arrive.
    </div>
    <br />
    <button
      className="ui small negative button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Unsubscribe
    </button>
  </div>
</div>
`;
